client/src/app/core/services/pdf-document.service.ts

```diff
@@ -31,10 +32,84 @@ export class PdfDocumentService {
      *
      * @param translate translations
      * @param configService read config values
+     * @param mediaManageService to read out font files as media data
      */
-    public constructor(private translate: TranslateService, private configService: ConfigService) {
-        // It should be possible to add own fonts here
-        pdfMake.vfs = pdfFonts.pdfMake.vfs;
+    public constructor(
+        private translate: TranslateService,
+        private configService: ConfigService,
+        private httpService: HttpService
+    ) {}
+
+    /**
+     * Define the pdfmake virtual file system for fonts
+     *
+     * @returns the vfs-object
+     */
+    private async getVfs(): Promise<object> {
+        const availableFonts = this.configService.instant('fonts_available');
+
+        const fontPathList: string[] = Array.from(
```

Comment:
Why is this necessary? You may give the type of the config object: `const availableFonts = this.configService.instant<string[]>('fonts_available');`, then no casting must be done.

Reply:
instant does not yet support templates. Currect syntax would be `const availableFonts = this.configService.instant('fonts_available') as string[]` or `const availableFonts: string[] = this.configService.instant('fonts_available');`

However:
new Set(myList) returns a "Set"-object without redundancies, which needs to be converted to an array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
Ok, can you add the generic parameter to instant? Just like get, then the interface in the config service is consistant..